src/main/java/com/hbm/world/gen/component/Component.java

```diff
@@ -663,7 +673,44 @@ protected void fillWithBlocks(World world, StructureBoundingBox box, int minX, i
 			}
 		}
 	}
-	
+	/** Have you ever wanted Glyphids in your building? No? Too bad! **/
+	protected void infest(World world, StructureBoundingBox box, int minX, int minY, int minZ, int maxX, int maxY, int maxZ, int maxSize, Random rand){
+		if(!enableInfestation)
+			return;
+
+		PollutionHandler.PollutionPerWorld ppw = PollutionHandler.perWorld.get(world);
```

Comment:
Wdym?

Reply:
To be fair, the infested buildings are not overly difficult, and the rewards mainly benefit early game

Reply:
glyphids are a force to be reckoned with early-game, and placing them directly in buildings only discourages their exploration until later (which kinda defeats the point).
martin's also right, but a big part of the 'new' structure system was to get rid of this kind of messy, sometimes cascade-causing worldgen

Reply:
I made the base infested building chance somewhat low, I didn't want to discourage their exploration, I wanted those buildings to encourage players to make early game guns which are often overlooked and gear and raid the buildings.
I made sure the payout was fairly worthwhile for an early game player

Reply:
> I wanted those buildings to encourage players to make early game guns

I really think the already common glyphid hives are reason enough.

Reply:
In hindsight, I did not fully account for what the old generation style could cause when smashed in with the more refined one. What I did account for was the cascading gen, via adding bounding box checks on the `infest` function. Structure hives are also much smaller given they are walless(to ensure glyphids could actually move through the building and out of it) so the chances of them going outside the alloted space for the structure is fairly slim

Reply:
I will remake the structure hive using the more refined system though, better safe than sorry

Reply:
Yeah but that doesn't really fix the concept issue, I think I'm gonna pass on this one.

Reply:
> already

Right now glyphids are an almost irrelevant threat, with their hives being able to get blocked off easily by terrain, and their only source of expansion is a scout which lazily meanders around.
Even in early game they are very easy to avoid.

Reply:
You can't please everyone, even if you put a giant neon sign saying evil bugs here, people would still die and complain

Reply:
There is a happy medium between decoration and unstoppable menace, and I worked for a fair while on finding it

Reply:
clueless

Reply:
I play tested these things for almost a month now and have been gathering feedback, it is possible to make glyphids interesting without making them pushovers or scape and run

Reply:
Which is why this is only part 1, in a series of PRs